src/jit/codegencommon.cpp

```diff
@@ -2465,6 +2465,21 @@ bool CodeGen::genCreateAddrMode(GenTreePtr  addr,
         return false;
     }
 
+#if defined(_TARGET_ARM_) && !defined(LEGACY_BACKEND)
+    if (mode == -1)
+    {
```

Comment:
I wonder why the `if (mode == -1)` is required. It seems like that would mean we would allow creating such address modes early, but then reject them when we go to actually generate them?